presto-native-execution/presto_cpp/main/PeriodicMemoryChecker.cpp

```diff
@@ -211,4 +214,17 @@ void PeriodicMemoryChecker::pushbackMemory() {
       kCounterMemoryPushbackLatencyMs, latencyMs * 1000);
   LOG(INFO) << "Shrunk " << velox::succinctBytes(freedBytes);
 }
+
+void startPeriodicMemoryChecker() {
```

Comment:
These methods were created in PeriodicMemoryChecker.cpp because so we could have some other MemoryCheckers that can extend from PeriodicMemoryChecker be instantiated and started here based on the machine's OS...

Reply:
> we could have some other MemoryCheckers that can extend from PeriodicMemoryChecker be instantiated

We would then have a factory that would return the corresponding implementation based on the config or machine' OS. 

See how the connector factories are implemented.

Reply:
I think following the DefaultBaseStatReporter and PrometheusStatsReporter approach is apt for this case.
Do we ever call the base PeriodicMemoryChecker today?

Reply:
@majetideepak No, we have never called or used it.